DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1CompareTrackSoA.cc

```diff
@@ -0,0 +1,200 @@
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+// DQM Histograming
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "CUDADataFormats/Track/interface/PixelTrackHeterogeneous.h"
+// for string manipulations
+#include <fmt/printf.h>
+
+class SiPixelPhase1CompareTrackSoA : public DQMEDAnalyzer {
+public:
+  explicit SiPixelPhase1CompareTrackSoA(const edm::ParameterSet&);
+  ~SiPixelPhase1CompareTrackSoA() override = default;
+  void bookHistograms(DQMStore::IBooker& ibooker, edm::Run const& iRun, edm::EventSetup const& iSetup) override;
+  void analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  const edm::EDGetTokenT<PixelTrackHeterogeneous> tokenSoATrackCPU_;
+  const edm::EDGetTokenT<PixelTrackHeterogeneous> tokenSoATrackGPU_;
+  const std::string topFolderName_;
+  const bool useQualityCut_;
+  const pixelTrack::Quality minQuality_;
+  const float dr2cut_;
+  MonitorElement* hnTracks;
+  MonitorElement* hnLooseAndAboveTracks;
+  MonitorElement* hnLooseAndAboveTracks_matched;
+  MonitorElement* hnHits;
+  MonitorElement* hnHitsVsPhi;
+  MonitorElement* hnHitsVsEta;
+  MonitorElement* hnLayers;
+  MonitorElement* hnLayersVsPhi;
+  MonitorElement* hnLayersVsEta;
+  MonitorElement* hchi2;
+  MonitorElement* hChi2VsPhi;
+  MonitorElement* hChi2VsEta;
+  MonitorElement* hpt;
+  MonitorElement* heta;
+  MonitorElement* hphi;
+  MonitorElement* hz;
+  MonitorElement* htip;
+  MonitorElement* hquality;
+};
+
+//
+// constructors
+//
+
+SiPixelPhase1CompareTrackSoA::SiPixelPhase1CompareTrackSoA(const edm::ParameterSet& iConfig)
+    : tokenSoATrackCPU_(consumes<PixelTrackHeterogeneous>(iConfig.getParameter<edm::InputTag>("pixelTrackSrcCPU"))),
+      tokenSoATrackGPU_(consumes<PixelTrackHeterogeneous>(iConfig.getParameter<edm::InputTag>("pixelTrackSrcGPU"))),
+      topFolderName_(iConfig.getParameter<std::string>("TopFolderName")),
+      useQualityCut_(iConfig.getParameter<bool>("useQualityCut")),
+      minQuality_(pixelTrack::qualityByName(iConfig.getParameter<std::string>("minQuality"))),
+      dr2cut_(iConfig.getParameter<double>("deltaR2cut")) {}
+
+//
+// -- Analyze
+//
+void SiPixelPhase1CompareTrackSoA::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  const auto& tsoaHandleCPU = iEvent.getHandle(tokenSoATrackCPU_);
+  const auto& tsoaHandleGPU = iEvent.getHandle(tokenSoATrackGPU_);
+  if (!tsoaHandleCPU.isValid() || !tsoaHandleGPU) {
+    edm::LogWarning("SiPixelPhase1CompareTrackSoA")
+        << "Either GPU or CPU tracks not found! Hence comparison not run!" << std::endl;
+    return;
+  }
+
+  auto const& tsoaCPU = *((tsoaHandleCPU.product())->get());
+  auto const& tsoaGPU = *((tsoaHandleGPU.product())->get());
```

Comment:
I _think_ these can be simplified to
```suggestion
  auto const& tsoaCPU = *tsoaHandleCPU;
  auto const& tsoaGPU = *tsoaHandleGPU;
```
?

Reply:
the `->get()` part is still needed, otherwise one gets compilation errors, but otherwise done at [76b4e67](https://github.com/cms-sw/cmssw/pull/37619/commits/76b4e679e9fa98f05f5fc7095d00eb4b03926ef5)